src/search/merge_and_shrink/labels.h

```diff
@@ -1,44 +1,63 @@
 #ifndef MERGE_AND_SHRINK_LABELS_H
 #define MERGE_AND_SHRINK_LABELS_H
 
+// TODO: rename this file to global_labels.h
+
 #include <memory>
 #include <vector>
 
 namespace merge_and_shrink {
-class Label {
-    /*
-      This class implements labels as used by merge-and-shrink transition systems.
-      Labels are opaque tokens that have an associated cost.
-    */
-    int cost;
+class GlobalLabelsConstIterator {
+    const std::vector<int> &label_costs;
+    std::size_t current_index;
+
+    void next_valid_index();
 public:
-    explicit Label(int cost_)
-        : cost(cost_) {
+    GlobalLabelsConstIterator(const std::vector<int> &label_costs, bool end);
+    void operator++();
+
+    int operator*() const {
+        return int(current_index);
     }
-    ~Label() {}
-    int get_cost() const {
-        return cost;
+
+    bool operator==(const GlobalLabelsConstIterator &rhs) const {
+        return current_index == rhs.current_index;
+    }
+
+    bool operator!=(const GlobalLabelsConstIterator &rhs) const {
+        return current_index != rhs.current_index;
     }
 };
 
 /*
-  This class serves both as a container class to handle the set of all labels
-  and to perform label reduction on this set.
+  This class serves both as a container class to handle the set of all global
+  labels and to perform label reduction on this set.
 */
-class Labels {
-    std::vector<std::unique_ptr<Label>> labels;
-    int max_size; // the maximum number of labels that can be created
+class GlobalLabels {
+    std::vector<int> label_costs;
+    int max_num_labels; // the maximum number of labels that can be created
 public:
-    explicit Labels(std::vector<std::unique_ptr<Label>> &&labels);
-    void reduce_labels(const std::vector<int> &old_label_nos);
-    bool is_current_label(int label_no) const;
-    int get_label_cost(int label_no) const;
+    GlobalLabels(std::vector<int> &&label_costs, int max_num_labels);
```

Comment:
`max_num_labels` sounds like something that should be the responsibility of this class rather than something that is passed in as a parameter. Can you compute it in the constructor based on the size of `label_costs`?

Reply:
See my answer at `fts_factory.cc`.